aleph/validation/schema/diagram.yml

```diff
@@ -0,0 +1,132 @@
+DiagramUpdate:
+  type: object
+  properties:
+    label:
+      type: string
+    summary:
+      type: string
+      nullable: true
+    layout:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/Layout'
+    entities:
+      type: array
+      items:
+        type: string
+  required: []
+
+DiagramCreate:
+  type: object
+  allOf:
+    - $ref: '#/components/schemas/DiagramUpdate'
+  properties:
+    collection_id:
+      type: integer
+  required: ["label", "collection_id"]
+
+Diagram:
+  type: object
+  allOf:
+  - $ref: '#/components/schemas/DatedModel'
+  properties:
+    collection:
+      type: object
+      allOf:
+      - $ref: '#/components/schemas/Collection'
+    role_id:
+      type: string
+    label:
+      type: string
+    summary:
+      type: string
+      nullable: true
+    entities:
+      type: array
+      items:
+        type: object
+        allOf:
+        - $ref: '#/components/schemas/EntityUpdate'
+        properties:
+          id:
+            type: string
+        required: ["id", "schema"]
+    layout:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/Layout'
+    writeable:
+      type: boolean
+  required: ["collection", "role_id", "label", "layout", "entities", "writeable"]
+
+Layout:
+  type: object
+  required: ["vertices", "edges"]
+  properties:
+    vertices:
+      type: array
+      items:
+        type: object
```

Comment:
Should we give these names? DiagramVertex, DiagramEdge, etc.?

Reply:
Done